sentry/src/main/java/io/sentry/SentryClient.java

```diff
@@ -496,6 +496,9 @@ public void captureSession(final @NotNull Session session, final @Nullable Objec
       if (sentryBaseEvent.getRequest() == null) {
         sentryBaseEvent.setRequest(scope.getRequest());
       }
+      if (sentryBaseEvent.getUser() == null) {
+        sentryBaseEvent.setUser(scope.getUser());
+      }
```

Comment:
what's about the combination with this check and `MainEventProcessor`? do they conflict with each other when `isSendDefaultPii` is enabled? I believe one of them will never be executed if it sets to a non-null value

Reply:
If user is not set on the event, `SentryClient` sets it from the scope. Then in `MainEventProcessor` if user is null it sets the user with a default ip address, if its not null but the ip address is null it sets the default ip address. What conflict do you mean?

Reply:
> @marandaneto for anyone who is not using integrations but does manual instrumentation this PR does provide a value. As I understand no matter how we solve the integrations, scope should be applied to transactions anyways.

I agree, but ideally this PR or another one should fix for our integrations too, see https://github.com/getsentry/sentry-java/issues/1335#issuecomment-814891298

Reply:
indeed, apply scope happens before event processors, that's why, the other way around would be problematic I believe.

Reply:
a problem that I see is, MainEventProcessor takes into consideration the `isSendDefaultPii` and sets IP address with `DEFAULT_IP_ADDRESS`, but that won't happen for transactions, this can be fixed later IMO when we run event processors for transactions too.

Reply:
We could start setting the default value even if a user is set, but the IpAddress is null.
I don't believe the "setting user semantics" dictates it's a all or nothing thing, or does it?

Can't we merge values instead? 

If I do `captureEvent(Event{ user = User(name))` while having `sendDefaultPii=true` I'd still like the the `{{auto}}` IP Address to be added to the user. I just set the `name` because that's all I have.